pkg/manager/manager.go

```diff
@@ -400,6 +404,100 @@ func New(config *rest.Config, options Options) (Manager, error) {
 	}, nil
 }
 
+// AndFrom will use a supplied type and convert to Options
+// any options already set on Options will be ignored, this is used to allow
+// cli flags to override anything specified in the config file
+func (o Options) AndFrom(loader config.DeferredFileLoader) (Options, error) {
```

Comment:
is there a reason this isn't just `ControllerConfiguration` instead?

Reply:
Because if we used `ControllerConfiguration` we wouldn't have a way to set the scheme on the loader if it was already provided by the `Options` that this is called on.

Reply:
Also following that we use the scheme to lazy load/decode the file which is why we don't just have `ControllerConfiguration` immediately.

Reply:
Alright, this has been updated, we can not use `ControllerConfiguration` to pass the `config.DeferredFileLoader` into it.